### ISSUE TITLE
startssl - certificates not working completely

### ISSUE THREAD

**Issue by Harper04** [nginx-proxy/nginx-proxy#207]
Hi i am using a StartSSL-Certificate which should be valid for www and non www domains.
The www.* domain is working like expected but the non-www returns the default ssl-certificate of the proxy.

**Comment by aindlq** [comment]
I think you need to provide more details. I'm successfully using free ones as well as verified wildcard certificates from StartSSL with nginx-proxy. Everything works out of the box without any issues.

**Comment by Harper04** [comment]
i tried
VIRTUAL_HOST=www.domain.com,domain.com
and added the same cert for both hosts with their respective names which did not work

using 
*.domain.com and CERT_NAME however does work

**Comment by avidsapp** [comment]
Can you be more specific? Is a wildcard certificate from StartSSL required with the CERT_NAME env variable? I'm having no luck with either the www or non-www domain and respective certificates out of the box, concatenated or separated from the ca.pem and sub.class1.server.sha2.ca.pem. Only error I can find on nginx container is [error] 57#57: *1658 upstream prematurely closed connection while reading response header from upstream

**Comment by aindlq** [comment]
Everything works fine for me out of the box. I just concatenated my .pem and sub.class1.server.sha2.ca.pem and then followed the instruction - https://github.com/jwilder/nginx-proxy#ssl-support . It works fine with free certificates as well as with wildcard certificates from StartSSL.

I do not use CERT_NAME, I use domain + .key, .cert approach. I also generated .dhparam.pem file, but obviously that it works fine without it as well.

I didn't try to use certificates for www.* domain.

**Comment by avidsapp** [comment]
My setup is similar to [this](http://serverfault.com/questions/642124/wordpress-behind-nginx-reverse-proxy-doesnt-work-in-https) with VIRTUAL_HOST=domain.com,www.domain.com (wp-config mod didn't work). Non-www domains redirect to www.*. I followed the readme instructions with both StartSSL and Comodo free certs on one domain and StartSSL on another as well. Wordpress isn't recognizing SSL on either, but http still works. [This didn't work either] (http://www.wpbeginner.com/wp-tutorials/how-to-add-ssl-and-https-in-wordpress/)

Port 443 exposed on both nginx-proxy and wordpress containers and "/path/to/certs" mounted in nginx-proxy. I'm using docker-compose for orchestration.

[Adding certs](http://www.jeedo.net/how-to-configure-and-install-free-ssl-certificates-from-startssl/) - I also tried sub.class1.server.sha2.ca.pem and sub.ckass1.server.ca.pem variations.

My thoughts are the proxy isn't serving the SSL certs, WordPress is conflicting with SSL, or docker-compose is conflicting.
